extern/sector-storage/ffiwrapper/sealer_cgo.go

```diff
@@ -530,9 +533,19 @@ func (sb *Sealer) SealPreCommit1(ctx context.Context, sector storage.SectorRef,
 	if err != nil {
 		return nil, xerrors.Errorf("presealing sector %d (%s): %w", sector.ID.Number, paths.Unsealed, err)
 	}
-	return p1o, nil
+
+	p1odec := map[string]interface{}{}
+	if err := json.Unmarshal(p1o, &p1odec); err != nil {
+		return nil, xerrors.Errorf("unmarshaling pc1 output: %w", err)
+	}
+
+	p1odec["_lotus_SealRandomness"] = ticket
```

Comment:
This should probably be a constant in this file or some such...also maybe rename it to indicate it's not real randomness, just for sanity-checking?

Reply:
Well, it is the real seal randomness (aka the ticket), we use this json thing to transport it into PC2 which doesn't have an easy way to get to it.